Userland/Libraries/LibCore/System.cpp

```diff
@@ -81,10 +81,33 @@ ErrorOr<void> pledge(StringView promises, StringView execpromises)
     HANDLE_SYSCALL_RETURN_VALUE("pledge", rc, {});
 }
 
+static bool dynamic_loader_unveiled { false };
+
+static ErrorOr<void> unveil_dynamic_loader()
+{
+    if (dynamic_loader_unveiled)
+        return {};
+    // FIXME: Try to find a way to not hardcode the dynamic loader path.
+    auto const dynamic_loader_path = "/usr/lib/Loader.so"sv;
+    auto const dynamic_loader_permissions = "x"sv;
+
+    Syscall::SC_unveil_params params {
+        { dynamic_loader_path.characters_without_null_termination(), dynamic_loader_path.length() },
+        { dynamic_loader_permissions.characters_without_null_termination(), dynamic_loader_permissions.length() },
+    };
+    int rc = syscall(SC_unveil, &params);
+    HANDLE_SYSCALL_RETURN_VALUE("unveil", rc, {});
+}
+
 ErrorOr<void> unveil(StringView path, StringView permissions)
 {
     auto const parsed_path = TRY(Core::SessionManagement::parse_path_with_sid(path));
 
+    if (permissions.contains('x')) {
+        TRY(unveil_dynamic_loader());
+        dynamic_loader_unveiled = true;
```

Comment:
I assume the intention for this to be set outside of `unveil_dynamic_loader` is to not set it if an error occurred? Is there even a point in having such kind of a retry mechanic as long as we always try to unveil the same path? In case there isn't, we could just scope that boolean inside of the `unveil_dynamic_loader` function and handle that entirely internally.